Oh no.
Error running app. If you need help, try the Streamlit docs and forums.

[     UTC     ] Logs for kpi-dashboard-vd.streamlit.app/

────────────────────────────────────────────────────────────────────────────────────────

[21:02:33] 🚀 Starting up repository: 'kpi-dashboard', branch: 'main', main module: 'app.py'

[21:02:33] 🐙 Cloning repository...

[21:02:33] 🐙 Cloning into '/mount/src/kpi-dashboard'...

[21:02:34] 🐙 Cloned repository!

[21:02:34] 🐙 Pulling code changes from Github...

[21:02:34] 📦 Processing dependencies...

Prepared 53 packages in 2.55s

Installed 53 packages in 148ms

 + altair==5.5.0

 + annotated-types==0.7.0

 + anyio==4.9.0

 + attrs==25.3.0

 + blinker==1.9.0

 + cachetools==6.1.0

 + certifi==2025.7.14

 + charset-normalizer==3.4.2

 + click==8.2.1

 + distro==1.9.0

 + et-xmlfile==2.0.0

 + gitdb==4.0.12

 + gitpython==3.1.44

 + h11==0.16.0

 + httpcore==1.0.9[2025-07-20 21:02:37.603652] 

 + httpx==0.28.1

 + idna==3.10

 + jinja2==3.1.6

 + jiter==0.10.0

 + jsonschema==4.24.0

 + jsonschema-specifications==2025.4.1

 + markupsafe==3.0.2

 + narwhals==[2025-07-20 21:02:37.603900] 1.47.0

 + numpy==2.3.1

 + openai==1.97.0

 + openpyxl==3.1.5

 + packaging==25.0

 + pandas==2.3.1

 + pillow==[2025-07-20 21:02:37.604113] 11.3.0

 + plotly==6.2.0

 + protobuf==6.31.1

 + pyarrow==20.0.0

 + pydantic==2.11.7

 + pydantic-core==2.33.2

 + pydeck==0.9.1

 + python-dateutil==2.9.0.post0

 + pytz==2025.2

 + referencing==0.36.2[2025-07-20 21:02:37.604295] 

 + requests==2.32.4

 + rpds-py==0.26.0

 + six==1.17.0

 + smmap==5.0.2

 + sniffio==1.3.1

 + streamlit==1.47.0[2025-07-20 21:02:37.604519] 

 + tenacity==9.1.2

 + toml==0.10.2

 + tornado==6.5.1

 + tqdm==4.67.1

 + typing-extensions==4.14.1

 + typing-inspection==0.4.1[2025-07-20 21:02:37.604686] 

 + tzdata==2025.2

 + urllib3==2.5.0

 + watchdog==6.0.0

[21:02:38] 🐍 Python dependencies were installed from /mount/src/kpi-dashboard/uv.lock using uv-sync.

[21:02:38] 📦 WARN: More than one requirements file detected in the repository. Available options: uv-sync /mount/src/kpi-dashboard/uv.lock, poetry /mount/src/kpi-dashboard/pyproject.toml. Used: uv-sync with /mount/src/kpi-dashboard/uv.lock

Check if streamlit is installed

Streamlit is already installed

[21:02:40] 📦 Processed dependencies!




[21:13:31] 🐙 Pulling code changes from Github...

[21:13:31] 📦 Processing dependencies...

[21:13:31] 📦 Processed dependencies!

2025-07-20 21:13:31.725 Uncaught app execution

Traceback (most recent call last):

  File "/home/adminuser/venv/lib/python3.13/site-packages/streamlit/runtime/scriptrunner/exec_code.py", line 128, in exec_func_with_error_handling

    result = func()

  File "/home/adminuser/venv/lib/python3.13/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 669, in code_to_exec

    exec(code, module.__dict__)  # noqa: S102

    ~~~~^^^^^^^^^^^^^^^^^^^^^^^

  File "/mount/src/kpi-dashboard/app.py", line 9, in <module>

    from database import Database

  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load

  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked

  File "<frozen importlib._bootstrap>", line 946, in _load_unlocked

KeyError: 'database'

  Stopping...




[21:13:46] 🔄 Updated app!

[21:13:46] ❗️ The service has encountered an error while checking the health of the Streamlit app: Get "http://localhost:8501/healthz": read tcp 127.0.0.1:34474->127.0.0.1:8501: read: connection reset by peer

main
rizvifarhan/kpi-dashboard/main/app.py


